artiq/frontend/artiq_client.py

```diff
@@ -298,9 +291,6 @@ def main():
             _show_dict(args, "devices", _show_devices)
         elif args.what == "datasets":
             _show_dict(args, "datasets", _show_datasets)
-        else:
```

Comment:
Better still raise an exception here to help catch program bugs. A simple ``raise ValueError`` will do.

Reply:
That is taken care of by `choices=[...]` in line 99:
https://github.com/m-labs/artiq/blob/b5a5be4ac8dfb10846ca413a91e8223429839197/artiq/frontend/artiq_client.py#L97-L100
https://docs.python.org/3/library/argparse.html#choices

Reply:
Sure. But if ``choices`` does not match this test (e.g. due to a typo) then there is a bug, and not raising an exception here will make it more obscure than it should be. Generally, it is good practice to always use the pattern ``if x=A: elif x=B: ... else: raise ValueError`` (short of a case statement in Python)

Reply:
Resolved in https://github.com/m-labs/artiq/pull/1185/commits/01e4d70dc1e7b2df035382a3a44a3a064999d096